Content.Client/Atmos/Monitor/UI/AirAlarmWindow.xaml.cs

```diff
@@ -0,0 +1,196 @@
+using System;
+using System.Collections.Generic;
+using Content.Client.Atmos.Monitor.UI.Widgets;
+using Content.Client.Message;
+using Content.Shared.Atmos;
+using Content.Shared.Atmos.Monitor;
+using Content.Shared.Atmos.Monitor.Components;
+using Content.Shared.Atmos.Piping.Unary.Components;
+using Content.Shared.Temperature;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Localization;
+
+namespace Content.Client.Atmos.Monitor.UI
+{
+    [GenerateTypedNameReferences]
+    public partial class AirAlarmWindow : SS14Window
+    {
+        public event Action<string, IAtmosDeviceData>? AtmosDeviceDataChanged;
+        public event Action<AtmosMonitorThresholdType, AtmosAlarmThreshold, Gas?>? AtmosAlarmThresholdChanged;
+        public event Action<AirAlarmMode>? AirAlarmModeChanged;
+        public event Action<string>? ResyncDeviceRequested;
+        public event Action? ResyncAllRequested;
```

Comment:
tbh these are all callbacks to the BUI, since calling these from another UI widget should lead it to raising an event that the BUI catches, and then causes it to send a message to the server with the relevant data

Reply:
if what you mean by this is just passing up all the data into the BUI itself and having the logic be performed there instead of down at the individual widget, then i get what you mean though